tests/integration/test_gomod_packages.py

```diff
@@ -0,0 +1,29 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import pytest
+
+import utils
+
+
+@pytest.mark.xfail(reason="Should fail until enabling gomod strict mode")
```

Comment:
Instead of failing here, could we have a new field in the `test_env_vars` file saying if the strict mode is active in that environment or not?

This way, it will be quite straightforward to toggle this test behavior based on environment configurations.

Then you would need to change the test to only perform the final assertion if the environment has strict mode enabled and in case it is disabled, the final request state should be `completed`.

Reply:
@athos-ribeiro Ok, I added a flag to `test_env_vars`.